src/Message.js

```diff
@@ -38,15 +38,11 @@ module.exports = class Message {
 	 * @param  {Object|Object[]}  [props.metadata]
 	 */
 	constructor (props) {
-		if (props.original !== parser.parse(props.original)) {
-			throw new Error('Cannot use dashes (-) in ' + props.original);
-		}
+		parser.parse(props.original);
```

Comment:
Let's catch and re-throw the error in both cases in order to include the full message: `Dashes are not allowed in identifiers, in message "${props.original}"`